### PR TITLE
Segment#8

### PR DESCRIPTION
**Pull request by Kamalpaneru**

### CODE REVIEW

**Comment by amitness** [comment]
One extra image is getting generated.

image

**Comment by Kamalpaneru** [comment]
I think that image is being corrupt somehow... I don't think its an extra generated image

**Comment by Kamalpaneru** [comment]
Add this segment too and provide it the path of the original video 'to segment.mp4'

selection_027

**Comment by eroj333** [comment]
We need a mechanism to check if a video exists or not. Should throw an error when not found.

**Comment by Kamalpaneru** [comment]
@eroj333 two ways to do it.. Either using ffmpeg or convert .mp4 to .mp3 and play it along the video using Converter.

File: segment/segment.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    out.release()
+
+def scale_video(video_location, dest_path="./out.mp4", scaleX=2, scaleY=2):
+    fps, frames = extract_frames(video_location)  # Choose your PathOut yourself.
+    scaled_frames = scale_frames(frames, scaleX, scaleY)
+    frames_to_video(scaled_frames, fps, dest_path)
+
+if __name__ == '__main__':
+    video_location = 'to_segment.mp4'
+    scale_video(video_location, dest_path="./out.mp4", scaleX=2, scaleY=2)
```
Comment (amitness) [inline segment/segment.py]: @Kamalpaneru scale_x and scale_y instead of camelCase. Follow snake_case.

File: segment/segment.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    for frame in frames:
+       size =  (frame.shape[1],frame.shape[0])
+       scaled_frame = cv2.resize(frame, (0,0), fx=scaleX, fy=scaleY, interpolation=cv2.INTER_CUBIC)
+       scaled_frames.append(scaled_frame)
+    return scaled_frames
+
+# Combine respective frames into a video
+def frames_to_video(frames, fps, pathOut="./out.mp4"):
+    print('Combining frames')
+    # Array of video frames path eg:'/home/kamal/Desktop/major-project/frames/frame1.png'
```
Comment (amitness) [inline segment/segment.py]: Remove comments once it works.

Comment (Kamalpaneru) [reply segment/segment.py]: Done

File: segment/segment.py (hunk 1 above)
Comment (Kamalpaneru) [reply segment/segment.py]: Done

Comment (Kamalpaneru) [reply segment/segment.py]: Wait .. I forgot to do it on all..Will be done shortly

**Pull request merged**